desafio-03/victorvssouza/javascript/main.js

```diff
@@ -0,0 +1,60 @@
+const readline = require("readline");
+
+function capturaDados(callback) {
+    const rl = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout
+    });
+
+    rl.question("Digite o valor inicial: ", (valorInicial) => {
+        rl.question("Digite o valor final: ", (valorFinal) => {
+            rl.close();
+            callback(Number(valorInicial), Number(valorFinal));
+        });
+    });
+}
+
+function validacaoDados(valorInicial, valorFinal) {
+    if (!Number.isInteger(valorInicial) || !Number.isInteger(valorFinal)) {
+        throw new Error("Os valores devem ser números inteiros.");
+    }
+    if (valorInicial < 0 || valorFinal < 0) {
+        throw new Error("Os valores devem ser números positivos.");
+    }
+    if (valorInicial > valorFinal) {
+        throw new Error("O valor final deve ser igual ou maior que o valor inicial.");
+    }
+}
+
+function validacaoPolindromo(numeroSelecionado) {
+    const stringNumeroSelecionado = numeroSelecionado.toString();
+    return stringNumeroSelecionado === stringNumeroSelecionado.split("").reverse().join("");
+}
+
+function buscaPolindromo(valorInicial, valorFinal) {
+    const palindromos = [];
+    for (let numero = valorInicial; numero <= valorFinal; numero++) {
+        if (validacaoPolindromo(numero)) {
+            palindromos.push(numero);
+        }
+    }
+    return palindromos;
+}
+
+function main() {
+    try {
+        capturaDados((valorInicial, valorFinal) => {
+            try {
+                validacaoDados(valorInicial, valorFinal);
+                const palindromos = buscaPolindromo(valorInicial, valorFinal);
+                console.log(`Números palíndromos entre ${valorInicial} e ${valorFinal}: ${palindromos.join(", ")}`);
+            } catch (error) {
+                console.error(error.message);
+            }
+        });
+    } catch (error) {
+        console.error(error.message);
+    }
+}
+
+main();
```

Comment:
```suggestion
if (require.main === module) {
    main();
}
```

Em NodeJS é assim que fazemos uma main. Fora isso, o desafio está ok!

Reply:
@victorvssouza Quando alguém abre uma issue você só marca ela quando resolvida nas seguintes circunstâncias:

1. A pessoa que solicitou as mudanças já revisou novamente o código e aprovou o PR.
2. A pessoa explicitamente falou que você pode fechar a issue.

Não é boa prática você mesmo marcar a issue como resolvida sem ser nesses casos acima; você espera uma nova revisão.

O que você deve fazer é solicitar o Review do seu código após fazer as alterações, mas mantenha as issues em aberto.

Reply:
OK!